torch/_export/constraints.py

```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+
+import torch._dynamo
+from torch._dynamo.comptime import comptime
+from torch.fx.experimental.symbolic_shapes import constrain_range
+
+
+# TODO: we want to hide this min/max stuff under some abstraction similar to
+# DynamicDim
+def add_inline_constraint(symbol, min: Optional[int] = None, max: Optional[int] = None):
+    """
+    Add min/max constraint on the intermediate symbol at tracing time
+    """
+
+    if torch._dynamo.is_compiling():
+        @comptime
+        def _(ctx):
+            node = ctx.get_local("symbol").as_proxy().node
+            min = ctx.get_local("min").as_python_constant()
+            max = ctx.get_local("max").as_python_constant()
+            constrain_range(node.meta["example_value"], min=min, max=max)
+            ctx.graph().call_function(constrain_range, (node,), {"min": min, "max": max})
+    else:
+        constrain_range(symbol, min=min, max=max)
+
+    return symbol
+
+def add_inline_size_constraint(symbol, min: int = 2, max: Optional[int] = None):
```

Comment:
I don't see the point of having two very similar functions exposed to the user. Pick one.

Reply:
One is for adding a constraint on a value you know will be used as the size of some other tensor -- this has to be >= 2 and therefore has an extra check. The other is for adding a constraint on just any value which I think can be any integer right?

Reply:
Yeah that was the intention (one for size constraints and other for any arbitrary symbol). We have seen a case where user is using the unbacked symint to do data-dependent slicing. For that, i think it is little weird to force the symbol value to be 2 because if user does something like `n - unbacked_val`, the shape can be >= 2 and we don't need to specialize.

Reply:
OK, may I suggest then that we use different names to make the distinction more explicit. For the first, `constrain_as_value` and the for the second, `constrain_as_size`. The words `add` and `inline` don't provide much information, so take them out.

Reply:
Yup! Those names sound better! Thanks!